api/keto/acl/v1/acl.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package keto.acl.v1;
+
+option go_package = "github.com/ory/keto/api/keto/acl/v1;acl";
+option csharp_namespace = "Ory.Keto.Acl.V1";
+option java_multiple_files = true;
+option java_outer_classname = "AclProto";
+option java_package = "sh.ory.keto.acl.v1";
+option php_namespace = "Ory\\Keto\\Acl\\V1";
+
+
+// RelationTuple relates an Object with a Subject.
+//
+// While a tuple reflects a relationship between Object
+// and Subject, they do not completely define the effective ACLs.
+message RelationTuple {
+  // The object related by this tuple.
+  Object object = 1;
+  // The relation between an Object and a Subject.
+  string relation = 2;
+  // The subject related by this tuple.
+  // A Subject either represents a concrete subject id or
+  // a SubjectSet that expands to more Subjects.
+  Subject subject = 3;
+  // Used to perform a consistent read-modify-write (lock).
+  string etag = 4;
```

Comment:
An ETAG ([**Resource freshness validation**](https://google.aip.dev/154)) is similar to a hash string of the current version of a RelationTuple including commit_time.
I basically tried to satisfy what is described in paper at `2.4.2 Write`, but maybe in a bit different way?

2.4.2
1) Talks about setting a lock on an object that clients can request in a Read so that in
2) the client does a Write only if that lock had not been changed on server side

I don't quite get it, since this could even result in an "almost infinite" loop between multiple clients competing for the same lock on an object and never manage to run 2 RPCs (read+write) in a row before another client captures the object's lock. Maybe you can help interpret the paper.


...I opted for similar way to support consistent read-modify-write, but **maybe I got lost there** :D
- When the client reads, the server provides the etag, that is the hash of the current RelationTuple snapshot.
- When the client does a write (e.g. delete) on that RelationTuple including the etag (unmodified, as given by server), the server checks that the passed etag equals to the etag computed using the stored RelationTuple to make sure the RelationTuple had not changed in the meantime.

Reply:
So your approach is not locking but rejecting `write` if the object changed between `read` and `write`? That is quite a good idea, as it assures the client that there was no change. On the other hand, tuples have no primary key. So updating one essentially means that we have to delete the old version and add the new version :thinking: I am not sure how to interpret the paper at that point, but I will think about it.